monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -75,77 +28,55 @@ class HTTPIslandAPIClient(IIslandAPIClient):
     def __init__(
         self,
         agent_event_serializer_registry: AgentEventSerializerRegistry,
-        retries=RETRIES,
     ):
         self._agent_event_serializer_registry = agent_event_serializer_registry
-        self._session = requests.Session()
-        retry_config = Retry(retries)
-        self._session.mount("https://", HTTPAdapter(max_retries=retry_config))
+        self.request_facade = HTTPRequestsFacade("")
```

Comment:
Why do this? Seems confusing why you'd pass in an empty string. Just make this `None` until we have the API URL.

Reply:
None doesn't have `get`, `post` and `put` methods. Even if you do `Optional[HTTPRequestsFacade]` mypy doesn't understand this and still complain. As I've said, the solution to this is to remove the temporal coupling, but I've already spent a lot of time improving this code and it's better than the one I found

Reply:
Then I'd say modify `HTTPRequestsFacade` to have the default `api_url` argument as `""` instead of passing it here.

Reply:
This won't solve anything, except instead of invalid facade we'll have invalid url. The actual solution is to solve temporal coupling. I've done it by raising exception if request methods are called on uninitialized HTTPClient